beangulp/importer.py

```diff
@@ -128,3 +128,55 @@ def file_date(self, file) -> Optional[date]:
           (If no date is returned, the file creation time is used. This is the
           default.)
         """
+
+
+class Importer:
+
+    @property
+    def name(self) -> str:
+        return f'{self.__class__.__module__}.{self.__class__.__name__}'
+
+    def identify(self, path: str) -> bool:
+        return False
+
+    def extract(self, path: str, existing: data.Entries = None) -> data.Entries:
```

Comment:
So we're not changing the interface here yet? :-)

Reply:
The first iteration of this patch removed the "existing" argument, but I later put it back: my importers use it. The use I have is to use the existing entries to train my ML transactions classifier. We could split this into a mandatoy `extract(filepath)` method and an optional `process(new, existing)` method (maybe with a better name). Or the post extraction processing could be moved to a method of the `Ingest` object that can be specialized in a subclass by the user. Or it could be moved to a completely new tool. However, for all these solutions, it wold not be possible to map the new interface to the old. It may be not that bad to provide compatibility only in a best effort way. Thought?